tests/phpunit/tests/multisite/siteMeta.php

```diff
@@ -250,14 +246,84 @@ public function test_query_update_site_meta_cache_true() {
 				)
 			);
 
-			$num_queries = $wpdb->num_queries;
+			$num_queries = get_num_queries();
 			get_site_meta( self::$site_id, 'foo', true );
-			$this->assertSame( $num_queries, $wpdb->num_queries );
+			$this->assertSame( 1, get_num_queries() - $num_queries );
 		}
 
-		public function test_query_update_site_meta_cache_false() {
-			global $wpdb;
+		/**
+		 * @ticket 58185
+		 */
+		public function test_lazy_load_site_meta() {
+			if ( ! is_site_meta_supported() ) {
+				$this->markTestSkipped( 'Test only runs with the blogmeta database table installed.' );
+			}
+
+			$filter = new MockAction();
+			add_filter( 'update_blog_metadata_cache', array( $filter, 'filter' ), 10, 2 );
+
+			$q = new WP_Site_Query(
+				array(
+					'ID' => self::$site_id,
+				)
+			);
+
+			$this->assertSameSets( array( (string) self::$site_id ), wp_list_pluck( $q->sites, 'blog_id' ), 'Site query should return the first test site' );
+
+			$q = new WP_Site_Query(
+				array(
+					'ID' => self::$site_id2,
+				)
+			);
+
+			$this->assertSameSets( array( (string) self::$site_id2 ), wp_list_pluck( $q->sites, 'blog_id' ), 'Site query should return the second test site' );
+
+			get_site_meta( self::$site_id2 );
+
+			$args     = $filter->get_args();
+			$first    = reset( $args );
+			$site_ids = end( $first );
+			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two site\'s meta' );
+		}
+
+		/**
+		 * @ticket 58185
+		 */
+		public function test_lazy_load_site_meta_fields_id() {
+			if ( ! is_site_meta_supported() ) {
+				$this->markTestSkipped( 'Test only runs with the blogmeta database table installed.' );
+			}
+
+			$filter = new MockAction();
+			add_filter( 'update_blog_metadata_cache', array( $filter, 'filter' ), 10, 2 );
+
+			$q = new WP_Site_Query(
+				array(
+					'ID'     => self::$site_id,
+					'fields' => 'ids',
+				)
+			);
+
+			$this->assertSameSets( array( self::$site_id ), $q->sites, 'Site query should return the first test site' );
+
+			$q = new WP_Site_Query(
+				array(
+					'ID'     => self::$site_id2,
+					'fields' => 'ids',
+				)
+			);
 
+			$this->assertSameSets( array( self::$site_id2 ), $q->sites, 'Site query should return the second test site' );
+
+			get_site_meta( self::$site_id2 );
+
+			$args     = $filter->get_args();
+			$first    = reset( $args );
+			$site_ids = end( $first );
+			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two site\'s meta' );
```

Comment:
```suggestion
			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two sites\' meta' );
```

Nit: sometimes I hate me too :D

Reply:
```suggestion
			$this->assertSameSets( $site_ids, array( self::$site_id, self::$site_id2 ), 'This should have two sites meta' );
```